### DISCUSSION TITLE
Custom Interception Scripts do not run

### DISCUSSION THREAD

**Discussion by tawaren** [JanssenProject/jans#6565] [Q&A] [answered]
I currently evaluate (play with) Janssen for a project to see if it is an option. But I struggle to get just a simple script running. I can create, install, and enable it (it shows up in the TUI and is marked as enabled). I did not get any compile errors or similar. However, nothing happens. I tried an UpdateToken script (in Java), but when a client requests a new token, the Log I placed there is not printed. I even made one throwing an exception to ensure I would see it running. Do I miss a crucial step?

**Comment by yuriyz** [comment]
Maybe you've run into bug which should be fixed today/tomorrow. https://github.com/JanssenProject/jans/issues/6561
Try any other script, e.g. `IntrospectionType` should be easy to try.

**Reply by ossdhaval** [reply]
@tawaren the issue mentioned by @yuriyz has been fixed-closed and delivered. It should be available in the latest nightly build. Can you confirm if it fixes the issue that you faced?

**Comment by tawaren** [comment]
It works now, but the problem was elsewhere. I used the wrong format to associate a script with a client. I eventually figured it out by looking at different example clients that had a script associated.  The documentation about this is sadly currently not there (unless I looked in the wrong place)

**Comment by nynymike** [comment]
Can you elaborate? We'd love to add this to the docs if it's missing! 

Or maybe you can make a PR for the docs :-)

**Reply by tawaren** [reply, marked as answer]
There is the page: https://docs.jans.io/v1.0.19/admin/auth-server/client-management/client-scripts/, which I assume would describe how to associate scripts with clients but is currently marked as "This content is in progress".  My problem was that when I configured the client (under Auth Server/Clients in the TUI) I entered for the "Client Scripts" configuration first just the name, then just the inum until I figured out that it expected a dn alla: "inum=...,ou=scripts,o=jans". If we decide to use Janssen (there are still other products we are evaluating to see which fits our usecase the best) I will consider writing a PR.

**Reply by ossdhaval** [reply]
Ah! We have [an open issue](https://github.com/JanssenProject/jans/issues/4866) to work on this document. We will prioritize this and at least put some basic information which would have helped you. Thanks for bringing this up. :+1:

**Reply by ossdhaval** [reply]
@tawaren Your concern is not addressed with [this fix](https://github.com/JanssenProject/jans/issues/6598#issuecomment-1825569344). Now TUI will offer a much better experience that is less error-prone. :rocket:

**Reply by tawaren** [reply]
Thanks, that's a very nice improvement and will make using custom scripts much more convenient.
